optuna/storages/_rdb/alembic/versions/v3.0.0.c.py

```diff
@@ -0,0 +1,136 @@
+"""Add intermediate_value_type column to represent +inf and -inf
+
+Revision ID: v3.0.0.c
+Revises: v3.0.0.b
+Create Date: 2022-05-16 17:17:28.810792
+
+"""
+import enum
+
+import numpy as np
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import orm
+
+
+# revision identifiers, used by Alembic.
+revision = "v3.0.0.c"
+down_revision = "v3.0.0.b"
+branch_labels = None
+depends_on = None
+
+
+BaseModel = declarative_base()
+RDB_MAX_FLOAT = np.finfo(np.float32).max
+RDB_MIN_FLOAT = np.finfo(np.float32).min
+
+
+class IntermediateValueModel(BaseModel):
+    class FloatTypeEnum(enum.Enum):
+        FINITE_OR_NAN = 1
+        INF_POS = 2
+        INF_NEG = 3
+
+    __tablename__ = "trial_intermediate_values"
+    trial_intermediate_value_id = sa.Column(sa.Integer, primary_key=True)
+    intermediate_value = sa.Column(sa.Float, nullable=True)
+    intermediate_value_type = sa.Column(sa.Enum(FloatTypeEnum), nullable=False)
+
+
+def _get_postgresql_float_type_enum():
+    from sqlalchemy.dialects import postgresql
+
+    return postgresql.ENUM("FINITE_OR_NAN", "INF_POS", "INF_NEG", name="floattypeenum")
+
+
+def upgrade():
+    bind = op.get_bind()
+
+    if bind.dialect.name == "postgresql":
+        _get_postgresql_float_type_enum().create(bind)
```

Comment:
`batch_op.add_column(sa.Column("...", sa.Enum(...)))` does not create Enum type on PostgreSQL, so that optuna-e2e script raised following errors.

```
$ docker compose run --rm optuna-300c bash src/upgrade.sh
mysql
...
postgresql
[I 2022-05-24 02:56:43,205] Upgrading the storage schema to the latest version.
[E 2022-05-24 02:56:43,465] (psycopg2.errors.UndefinedObject) type "floattypeenum" does not exist
LINE 1: ...mediate_values ADD COLUMN intermediate_value_type floattypee...
                                                             ^

[SQL: ALTER TABLE trial_intermediate_values ADD COLUMN intermediate_value_type floattypeenum DEFAULT 'FINITE_OR_NAN' NOT NULL]
(Background on this error at: https://sqlalche.me/e/14/f405)
...
```

<details>

<summary>list of Enum types</summary>

```
optuna=# alter table trial_intermediate_values add column intermediate_value_type floattypeenum default 'FINITE_OR_NAN' not null;
ERROR:  type "floattypeenum" does not exist
LINE 1: ...mediate_values add column intermediate_value_type floattypee...
                                                             ^
optuna=# select n.nspname as enum_schema,
       t.typname as enum_name,
       e.enumlabel as enum_value
from pg_type t
   join pg_enum e on t.oid = e.enumtypid
   join pg_catalog.pg_namespace n ON n.oid = t.typnamespace;
 enum_schema |   enum_name    | enum_value
-------------+----------------+------------
 public      | studydirection | NOT_SET
 public      | studydirection | MINIMIZE
 public      | studydirection | MAXIMIZE
 public      | trialstate     | RUNNING
 public      | trialstate     | COMPLETE
 public      | trialstate     | PRUNED
 public      | trialstate     | FAIL
 public      | trialstate     | WAITING

(8 rows)
```

</details>

According to the [SQLAlchemy's documentation](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#sqlalchemy.dialects.postgresql.ENUM), create/drop behavior of ENUM is necessarily intricate. In fact, when using Enum type in PostgreSQL, it is required to create Enum type in separated SQL as not like MySQL.

```
# PostgreSQL
CREATE TYPE floattypeenum AS ENUM ('FINITE_OR_NAN', 'INF_POS', 'INF_NEG')
ALTER TABLE trial_intermediate_values ADD COLUMN intermediate_value_type floattypeenum DEFAULT 'FINITE_OR_NAN' NOT NULL
```

```
# MySQL
ALTER TABLE trial_intermediate_values ADD COLUMN intermediate_value_type ENUM('FINITE_OR_NAN','INF_POS','INF_NEG') NOT NULL DEFAULT 'FINITE_OR_NAN'
```

For these reasons, our migration script requires the postgresql specific logics. As an alternative approach, we can use VARCHAR column with CHECK constraints by using [native_enum=False](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Enum.params.native_enum) and [create_constraint=True](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Enum.params.create_constraint) options.

Reply:
@toshihikoyanase suggested me to use `sa.Enum(...).create()` method. Thanks! I'll try to use this.

* https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Enum.create
* https://github.com/sqlalchemy/alembic/issues/278#issuecomment-904343421

Reply:
I confirmed that `sq.Enum(...).create()` works as expected. Full output of `bash src/upgrade.sh` of optuna-e2e with SQLAlchemy's INFO logging is here.

```
# bash src/upgrade.sh
mysql
[I 2022-05-24 08:29:01,607] Upgrading the storage schema to the latest version.
[I 2022-05-24 08:29:01,633] SELECT DATABASE()
[I 2022-05-24 08:29:01,634] [raw sql] {}
[I 2022-05-24 08:29:01,637] SELECT @@sql_mode
[I 2022-05-24 08:29:01,638] [raw sql] {}
[I 2022-05-24 08:29:01,640] SELECT @@lower_case_table_names
[I 2022-05-24 08:29:01,641] [raw sql] {}
[I 2022-05-24 08:29:01,647] SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = %(table_schema)s AND table_name = %(table_name)s
[I 2022-05-24 08:29:01,648] [generated in 0.00093s] {'table_schema': 'optuna', 'table_name': 'alembic_version'}
[I 2022-05-24 08:29:01,657] SELECT alembic_version.version_num
FROM alembic_version
[I 2022-05-24 08:29:01,658] [generated in 0.00111s] {}
[I 2022-05-24 08:29:01,752] BEGIN (implicit)
[I 2022-05-24 08:29:01,757] ALTER TABLE trial_intermediate_values ADD COLUMN intermediate_value_type ENUM('FINITE_OR_NAN','INF_POS','INF_NEG') NOT NULL DEFAULT 'FINITE_OR_NAN'
[I 2022-05-24 08:29:01,758] [no key 0.00076s] {}
[I 2022-05-24 08:29:01,801] ALTER TABLE trial_intermediate_values ALTER COLUMN intermediate_value_type DROP DEFAULT
[I 2022-05-24 08:29:01,801] [no key 0.00092s] {}
[I 2022-05-24 08:29:01,830] SELECT trial_intermediate_values.trial_intermediate_value_id AS trial_intermediate_values_trial_intermediate_value_id, trial_intermediate_values.intermediate_value AS trial_intermediate_values_intermediate_value, trial_intermediate_values.intermediate_value_type AS trial_intermediate_values_intermediate_value_type
FROM trial_intermediate_values
[I 2022-05-24 08:29:01,830] [generated in 0.00101s] {}
[I 2022-05-24 08:29:01,869] UPDATE trial_intermediate_values SET intermediate_value_type=%(intermediate_value_type)s WHERE trial_intermediate_values.trial_intermediate_value_id = %(trial_intermediate_values_trial_intermediate_value_id)s
[I 2022-05-24 08:29:01,870] [generated in 0.00169s] ({'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 1}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 2}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 4}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 5}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 7}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 8}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 10}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 11}  ... displaying 10 of 20 total bound parameter sets ...  {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 28}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 29})
[I 2022-05-24 08:29:01,903] UPDATE alembic_version SET version_num='v3.0.0.c' WHERE alembic_version.version_num = 'v3.0.0.b'
[I 2022-05-24 08:29:01,904] [generated in 0.00104s] {}
[I 2022-05-24 08:29:01,907] COMMIT
[I 2022-05-24 08:29:01,914] BEGIN (implicit)
[I 2022-05-24 08:29:01,916] SELECT version_info.version_info_id AS version_info_version_info_id, version_info.schema_version AS version_info_schema_version, version_info.library_version AS version_info_library_version
FROM version_info
[I 2022-05-24 08:29:01,916] [cached since 0.8519s ago] {}
[I 2022-05-24 08:29:01,923] COMMIT
[I 2022-05-24 08:29:01,926] Completed to upgrade the storage.
postgresql
[I 2022-05-24 08:29:05,792] Upgrading the storage schema to the latest version.
[I 2022-05-24 08:29:05,880] select pg_catalog.version()
[I 2022-05-24 08:29:05,881] [raw sql] {}
[I 2022-05-24 08:29:05,886] select current_schema()
[I 2022-05-24 08:29:05,886] [raw sql] {}
[I 2022-05-24 08:29:05,890] show standard_conforming_strings
[I 2022-05-24 08:29:05,891] [raw sql] {}
[I 2022-05-24 08:29:05,894] BEGIN (implicit)
[I 2022-05-24 08:29:05,896] select relname from pg_class c join pg_namespace n on n.oid=c.relnamespace where pg_catalog.pg_table_is_visible(c.oid) and relname=%(name)s
[I 2022-05-24 08:29:05,897] [generated in 0.00083s] {'name': 'alembic_version'}
[I 2022-05-24 08:29:05,905] SELECT alembic_version.version_num
FROM alembic_version
[I 2022-05-24 08:29:05,906] [generated in 0.00118s] {}
[I 2022-05-24 08:29:05,994]
            SELECT EXISTS (
                SELECT * FROM pg_catalog.pg_type t
                WHERE t.typname = %(typname)s
                AND pg_type_is_visible(t.oid)
                )

[I 2022-05-24 08:29:05,995] [generated in 0.00105s] {'typname': 'floattypeenum'}
[I 2022-05-24 08:29:06,003] CREATE TYPE floattypeenum AS ENUM ('FINITE_OR_NAN', 'INF_POS', 'INF_NEG')
[I 2022-05-24 08:29:06,003] [no key 0.00078s] {}
[I 2022-05-24 08:29:06,018] ALTER TABLE trial_intermediate_values ADD COLUMN intermediate_value_type floattypeenum DEFAULT 'FINITE_OR_NAN' NOT NULL
[I 2022-05-24 08:29:06,019] [no key 0.00086s] {}
[I 2022-05-24 08:29:06,031] ALTER TABLE trial_intermediate_values ALTER COLUMN intermediate_value_type DROP DEFAULT
[I 2022-05-24 08:29:06,032] [no key 0.00104s] {}
[I 2022-05-24 08:29:06,045] SELECT trial_intermediate_values.trial_intermediate_value_id AS trial_intermediate_values_trial_intermediate_value_id, trial_intermediate_values.intermediate_value AS trial_intermediate_values_intermediate_value, trial_intermediate_values.intermediate_value_type AS trial_intermediate_values_intermediate_value_type
FROM trial_intermediate_values
[I 2022-05-24 08:29:06,046] [generated in 0.00110s] {}
[I 2022-05-24 08:29:06,080] UPDATE trial_intermediate_values SET intermediate_value_type=%(intermediate_value_type)s WHERE trial_intermediate_values.trial_intermediate_value_id = %(trial_intermediate_values_trial_intermediate_value_id)s
[I 2022-05-24 08:29:06,081] [generated in 0.00179s] ({'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 1}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 2}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 4}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 5}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 7}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 8}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 10}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 11}  ... displaying 10 of 20 total bound parameter sets ...  {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 28}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 29})
[I 2022-05-24 08:29:06,106] UPDATE alembic_version SET version_num='v3.0.0.c' WHERE alembic_version.version_num = 'v3.0.0.b'
[I 2022-05-24 08:29:06,107] [generated in 0.00107s] {}
[I 2022-05-24 08:29:06,113] COMMIT
[I 2022-05-24 08:29:06,119] BEGIN (implicit)
[I 2022-05-24 08:29:06,120] SELECT version_info.version_info_id AS version_info_version_info_id, version_info.schema_version AS version_info_schema_version, version_info.library_version AS version_info_library_version
FROM version_info
[I 2022-05-24 08:29:06,121] [cached since 0.7972s ago] {}
[I 2022-05-24 08:29:06,126] COMMIT
[I 2022-05-24 08:29:06,128] Completed to upgrade the storage.
sqlite
[I 2022-05-24 08:29:09,960] Upgrading the storage schema to the latest version.
[I 2022-05-24 08:29:09,984] PRAGMA main.table_info("alembic_version")
[I 2022-05-24 08:29:09,985] [raw sql] ()
[I 2022-05-24 08:29:09,995] SELECT alembic_version.version_num
FROM alembic_version
[I 2022-05-24 08:29:09,996] [generated in 0.00098s] ()
[I 2022-05-24 08:29:10,091] BEGIN (implicit)
[I 2022-05-24 08:29:10,096] ALTER TABLE trial_intermediate_values ADD COLUMN intermediate_value_type VARCHAR(13) DEFAULT 'FINITE_OR_NAN' NOT NULL
[I 2022-05-24 08:29:10,097] [no key 0.00079s] ()
[I 2022-05-24 08:29:10,138] PRAGMA main.table_info("trial_intermediate_values")
[I 2022-05-24 08:29:10,139] [raw sql] ()
[I 2022-05-24 08:29:10,151] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,152] [raw sql] ('trial_intermediate_values',)
[I 2022-05-24 08:29:10,160] PRAGMA main.foreign_key_list("trial_intermediate_values")
[I 2022-05-24 08:29:10,161] [raw sql] ()
[I 2022-05-24 08:29:10,166] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,166] [raw sql] ('trial_intermediate_values',)
[I 2022-05-24 08:29:10,182] PRAGMA main.table_info("trials")
[I 2022-05-24 08:29:10,183] [raw sql] ()
[I 2022-05-24 08:29:10,191] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,192] [raw sql] ('trials',)
[I 2022-05-24 08:29:10,197] PRAGMA main.foreign_key_list("trials")
[I 2022-05-24 08:29:10,197] [raw sql] ()
[I 2022-05-24 08:29:10,202] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,204] [raw sql] ('trials',)
[I 2022-05-24 08:29:10,210] PRAGMA main.table_info("studies")
[I 2022-05-24 08:29:10,211] [raw sql] ()
[I 2022-05-24 08:29:10,217] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,218] [raw sql] ('studies',)
[I 2022-05-24 08:29:10,222] PRAGMA main.foreign_key_list("studies")
[I 2022-05-24 08:29:10,223] [raw sql] ()
[I 2022-05-24 08:29:10,225] PRAGMA temp.foreign_key_list("studies")
[I 2022-05-24 08:29:10,225] [raw sql] ()
[I 2022-05-24 08:29:10,227] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,228] [raw sql] ('studies',)
[I 2022-05-24 08:29:10,232] PRAGMA main.index_list("studies")
[I 2022-05-24 08:29:10,233] [raw sql] ()
[I 2022-05-24 08:29:10,238] PRAGMA main.index_info("ix_studies_study_name")
[I 2022-05-24 08:29:10,239] [raw sql] ()
[I 2022-05-24 08:29:10,244] PRAGMA main.index_list("studies")
[I 2022-05-24 08:29:10,245] [raw sql] ()
[I 2022-05-24 08:29:10,249] PRAGMA main.index_info("ix_studies_study_name")
[I 2022-05-24 08:29:10,250] [raw sql] ()
[I 2022-05-24 08:29:10,254] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,254] [raw sql] ('studies',)
[I 2022-05-24 08:29:10,270] PRAGMA main.index_list("trials")
[I 2022-05-24 08:29:10,271] [raw sql] ()
[I 2022-05-24 08:29:10,275] PRAGMA temp.index_list("trials")
[I 2022-05-24 08:29:10,276] [raw sql] ()
[I 2022-05-24 08:29:10,278] PRAGMA main.index_list("trials")
[I 2022-05-24 08:29:10,278] [raw sql] ()
[I 2022-05-24 08:29:10,283] PRAGMA temp.index_list("trials")
[I 2022-05-24 08:29:10,283] [raw sql] ()
[I 2022-05-24 08:29:10,285] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,286] [raw sql] ('trials',)
[I 2022-05-24 08:29:10,292] PRAGMA main.index_list("trial_intermediate_values")
[I 2022-05-24 08:29:10,293] [raw sql] ()
[I 2022-05-24 08:29:10,298] PRAGMA main.index_list("trial_intermediate_values")
[I 2022-05-24 08:29:10,299] [raw sql] ()
[I 2022-05-24 08:29:10,303] PRAGMA main.index_info("sqlite_autoindex_trial_intermediate_values_1")
[I 2022-05-24 08:29:10,304] [raw sql] ()
[I 2022-05-24 08:29:10,308] SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
[I 2022-05-24 08:29:10,309] [raw sql] ('trial_intermediate_values',)
[I 2022-05-24 08:29:10,323]
CREATE TABLE _alembic_tmp_trial_intermediate_values (
	trial_intermediate_value_id INTEGER NOT NULL,
	trial_id INTEGER NOT NULL,
	step INTEGER NOT NULL,
	intermediate_value FLOAT,
	intermediate_value_type VARCHAR(13) NOT NULL,
	PRIMARY KEY (trial_intermediate_value_id),
	UNIQUE (trial_id, step),
	FOREIGN KEY(trial_id) REFERENCES trials (trial_id)
)


[I 2022-05-24 08:29:10,324] [no key 0.00097s] ()
[I 2022-05-24 08:29:10,359] INSERT INTO _alembic_tmp_trial_intermediate_values (trial_intermediate_value_id, trial_id, step, intermediate_value, intermediate_value_type) SELECT trial_intermediate_values.trial_intermediate_value_id, trial_intermediate_values.trial_id, trial_intermediate_values.step, trial_intermediate_values.intermediate_value, trial_intermediate_values.intermediate_value_type
FROM trial_intermediate_values
[I 2022-05-24 08:29:10,360] [generated in 0.00112s] ()
[I 2022-05-24 08:29:10,388]
DROP TABLE trial_intermediate_values
[I 2022-05-24 08:29:10,389] [no key 0.00109s] ()
[I 2022-05-24 08:29:10,408] ALTER TABLE _alembic_tmp_trial_intermediate_values RENAME TO trial_intermediate_values
[I 2022-05-24 08:29:10,409] [no key 0.00098s] ()
[I 2022-05-24 08:29:10,424] SELECT trial_intermediate_values.trial_intermediate_value_id AS trial_intermediate_values_trial_intermediate_value_id, trial_intermediate_values.intermediate_value AS trial_intermediate_values_intermediate_value, trial_intermediate_values.intermediate_value_type AS trial_intermediate_values_intermediate_value_type
FROM trial_intermediate_values
[I 2022-05-24 08:29:10,425] [generated in 0.00096s] ()
[I 2022-05-24 08:29:10,456] UPDATE trial_intermediate_values SET intermediate_value_type=? WHERE trial_intermediate_values.trial_intermediate_value_id = ?
[I 2022-05-24 08:29:10,456] [generated in 0.00163s] (('INF_POS', 1), ('INF_NEG', 2), ('INF_POS', 4), ('INF_NEG', 5), ('INF_POS', 7), ('INF_NEG', 8), ('INF_POS', 10), ('INF_NEG', 11)  ... displaying 10 of 20 total bound parameter sets ...  ('INF_POS', 28), ('INF_NEG', 29))
[I 2022-05-24 08:29:10,464] UPDATE alembic_version SET version_num='v3.0.0.c' WHERE alembic_version.version_num = 'v3.0.0.b'
[I 2022-05-24 08:29:10,465] [generated in 0.00095s] ()
[I 2022-05-24 08:29:10,479] COMMIT
[I 2022-05-24 08:29:10,500] BEGIN (implicit)
[I 2022-05-24 08:29:10,502] SELECT version_info.version_info_id AS version_info_version_info_id, version_info.schema_version AS version_info_schema_version, version_info.library_version AS version_info_library_version
FROM version_info
[I 2022-05-24 08:29:10,503] [cached since 1.084s ago] ()
[I 2022-05-24 08:29:10,518] COMMIT
[I 2022-05-24 08:29:10,520] Completed to upgrade the storage.
mssql
[I 2022-05-24 08:29:14,362] Upgrading the storage schema to the latest version.
[I 2022-05-24 08:29:14,403] select @@version
[I 2022-05-24 08:29:14,404] [raw sql] {}
[I 2022-05-24 08:29:14,407] SELECT schema_name()
[I 2022-05-24 08:29:14,408] [generated in 0.00091s] {}
[I 2022-05-24 08:29:14,414] SELECT CAST('test max support' AS NVARCHAR(max))
[I 2022-05-24 08:29:14,415] [generated in 0.00109s] {}
[I 2022-05-24 08:29:14,421] BEGIN (implicit)
[I 2022-05-24 08:29:14,429] SELECT [INFORMATION_SCHEMA].[TABLES].[TABLE_NAME]
FROM [INFORMATION_SCHEMA].[TABLES]
WHERE [INFORMATION_SCHEMA].[TABLES].[TABLE_TYPE] = CAST(%(table_type_1)s AS NVARCHAR(max)) AND [INFORMATION_SCHEMA].[TABLES].[TABLE_NAME] = CAST(%(table_name_1)s AS NVARCHAR(max)) AND [INFORMATION_SCHEMA].[TABLES].[TABLE_SCHEMA] = CAST(%(table_schema_1)s AS NVARCHAR(max))
[I 2022-05-24 08:29:14,430] [generated in 0.00132s] {'table_type_1': 'BASE TABLE', 'table_name_1': 'alembic_version', 'table_schema_1': 'dbo'}
[I 2022-05-24 08:29:14,436] SELECT alembic_version.version_num
FROM alembic_version
[I 2022-05-24 08:29:14,436] [generated in 0.00113s] {}
[I 2022-05-24 08:29:14,531] ALTER TABLE trial_intermediate_values ADD intermediate_value_type VARCHAR(13) NOT NULL DEFAULT 'FINITE_OR_NAN'
[I 2022-05-24 08:29:14,532] [no key 0.00087s] {}
[I 2022-05-24 08:29:14,538] declare @const_name varchar(256)
select @const_name = [name] from sys.default_constraints
where parent_object_id = object_id('trial_intermediate_values')
and col_name(parent_object_id, parent_column_id) = 'intermediate_value_type'
exec('alter table trial_intermediate_values drop constraint ' + @const_name)
[I 2022-05-24 08:29:14,538] [no key 0.00082s] {}
[I 2022-05-24 08:29:14,563] SELECT trial_intermediate_values.trial_intermediate_value_id AS trial_intermediate_values_trial_intermediate_value_id, trial_intermediate_values.intermediate_value AS trial_intermediate_values_intermediate_value, trial_intermediate_values.intermediate_value_type AS trial_intermediate_values_intermediate_value_type
FROM trial_intermediate_values
[I 2022-05-24 08:29:14,564] [generated in 0.00102s] {}
[I 2022-05-24 08:29:14,596] UPDATE trial_intermediate_values SET intermediate_value_type=%(intermediate_value_type)s WHERE trial_intermediate_values.trial_intermediate_value_id = %(trial_intermediate_values_trial_intermediate_value_id)s
[I 2022-05-24 08:29:14,597] [generated in 0.00168s] ({'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 1}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 2}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 4}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 5}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 7}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 8}, {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 10}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 11}  ... displaying 10 of 20 total bound parameter sets ...  {'intermediate_value_type': 'INF_POS', 'trial_intermediate_values_trial_intermediate_value_id': 28}, {'intermediate_value_type': 'INF_NEG', 'trial_intermediate_values_trial_intermediate_value_id': 29})
[I 2022-05-24 08:29:14,616] UPDATE alembic_version SET version_num='v3.0.0.c' WHERE alembic_version.version_num = 'v3.0.0.b'
[I 2022-05-24 08:29:14,617] [generated in 0.00098s] {}
[I 2022-05-24 08:29:14,621] COMMIT
[I 2022-05-24 08:29:14,630] BEGIN (implicit)
[I 2022-05-24 08:29:14,631] SELECT version_info.version_info_id AS version_info_version_info_id, version_info.schema_version AS version_info_schema_version, version_info.library_version AS version_info_library_version
FROM version_info
[I 2022-05-24 08:29:14,632] [cached since 0.8021s ago] {}
[I 2022-05-24 08:29:14,638] COMMIT
[I 2022-05-24 08:29:14,641] Completed to upgrade the storage.
```